huaweicloud/resource_huaweicloud_identity_agency_test.go

```diff
@@ -0,0 +1,141 @@
+package huaweicloud
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/huaweicloud/golangsdk/openstack/identity/v3/agency"
+)
+
+func TestAccIdentityAgency_basic(t *testing.T) {
+	var agency agency.Agency
+
+	rName := fmt.Sprintf("tf-acc-test-%s", acctest.RandString(5))
+	resourceName := "huaweicloud_identity_agency.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckIdentityAgencyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccIdentityAgency_basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckIdentityAgencyExists(resourceName, &agency),
+					resource.TestCheckResourceAttr(resourceName, "name", rName),
```

Comment:
suggest to add more attr check

Reply:
now we have five attribute check in acc test.